src/methods/aaft.jl

```diff
@@ -16,14 +17,76 @@ monotonic nonlinear transformation of a linear Gaussian process
 struct AAFT <: Surrogate end
 
 function surrogenerator(x, method::AAFT, rng = Random.default_rng())
-    init = surrogenerator(x, RandomFourier(true), rng)
-    return SurrogateGenerator(method, x, init, rng)
+    n = length(x)
+
+    # Forward Fourier transform plan
+    forward = plan_rfft(x)
+
+    m = mean(x)
+
+    𝓕 = forward * (x .- m)
+
+    init = (
+        # A sorted version of `x`. Used when rescaling back to original values.
+        x_sorted = sort(x),
+
+        # A vector that holds permutation indices
+        ix = zeros(Int, n),
+        
+        # Inverse Fourier transform plan
+        inverse = plan_irfft(forward * x, n),
+
+        # Mean of `x`
+        m = m,
+
+        # Forward transform on mean-subtracted data
+        𝓕 = 𝓕,
+
+        # The following variables are pre-computed here, so we don't re-allocate 
+        # when calling the surrogate generator.
+        # ----------------------------------------------------------------------
+        r = abs.(𝓕),
+
+        # Phases  (compute here, so we don't allocate when generating)
+        ϕ = angle.(𝓕),
+
+        # Holds the new transform (after shuffling phases/amplitudes)
+        shuffled𝓕 = similar(𝓕),
+
+        # Randomized coefficients,
+        coeffs = zero(𝓕),
+
+        n = n,
+    )
+
+    return SurrogateGenerator(method, x, similar(x), init, rng)
 end
 
-function (rf::SurrogateGenerator{<:AAFT})()
-    x = rf.x
-    xs = sort(x)
-    s = rf.init()
-    s[sortperm(s)] .= xs
+function (sg::SurrogateGenerator{<:AAFT})()
+    # Initialization data
+    s, rng = sg.s, sg.rng
+
+    init_fields = (:x_sorted, :ix, :inverse, :m, :r, :ϕ, :shuffled𝓕, :coeffs, :n)
+        x_sorted, ix,  inverse, m, r, ϕ, shuffled𝓕, coeffs, n = 
+        getfield.(Ref(sg.init), init_fields)
+
+    # A Fourier surrogate is the starting point for the AAFT surrogate. Generate 
+    # one such surrogate and assign it to `sg.s`.
+    
+    # `coeffs` := randomized phases
+    coeffs .= rand(rng, Uniform(0, 2π), length(shuffled𝓕))
+
+    # Updated Fourier transform, with shuffled phases
+    shuffled𝓕 .= r .* exp.(coeffs .* 1im)
+    
+    # Inverse Fourier transform
+    s .= (inverse * shuffled𝓕) .+ m
```

Comment:
Personally, I'm not a fan of comments like these. And in fact, the same opinion is stated in books that teach how to write elegant code. Comments like these provides no real information. They tell you what the line of code says by itself. So the comments inflate the total amount of code and make it harder to review and navigate, but its not like they help the user understand by providing more information. 

Example:
```julia
# square y and then get logarithm for the answer
answer = log(y^2)
```

Another thing that makes the code difficult to read is that it has too many breaks in empty lines. Similar to a paper, contiguous processes form a paragraph. They don't have empty lines that interrupt them. Instead, empty lines separate different paragraphs. 

If you have some spare time, have a look in this workshop I gave about writing good code: https://youtu.be/iIIBFujl254 it uses information from various sources and own experience.

Reply:
Thanks for the link! I'll at least remove redundant code from the PR. 

> _"proper use of comments is to compensate for our failure  to express ourselves in code"_

Oh yes, I feel that. 😁

Reply:
hahahahahahah yeah I know!!!!